### PR TITLE
feat(simplified homepage): adding option to show limited entity types on homepage

### PR DESCRIPTION
**Pull request by gabe-lyons**
Adds a new setting to optionally show a limited homepage:
image

Can be managed from users preferences:
image

Can also be set default on for instances with the SHOW_SIMPLIFIED_HOMEPAGE_BY_DEFAULT feature flag set to true.

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Review by jjoyce0510 — COMMENTED** [review]
Let's call "User Settings" as "Appearance" (since this pertains to appearance)

**Comment by gabe-lyons** [comment]
@jjoyce0510 updated!

File: datahub-graphql-core/src/main/resources/entity.graphql [hunk 1]
```diff
@@ -493,6 +493,11 @@ type Mutation {
     Updates the soft deleted status for a batch of assets
     """
     batchUpdateSoftDeleted(input: BatchUpdateSoftDeletedInput!): Boolean
+
+    """
+    Update a user setting
+    """
+    updateUserSetting(input: UpdateUserSettingsInput!): Boolean
```
Comment (jjoyce0510) [inline datahub-graphql-core/src/main/resources/entity.graphql:500]: settings?

File: datahub-graphql-core/src/main/resources/entity.graphql [hunk 2]
```diff
@@ -9059,3 +9080,26 @@ input BatchUpdateSoftDeletedInput {
...
+    Show simplified homepage
+    """
+    SHOW_SIMPLIFIED_HOMEPAGE
+}
+
+
+"""
+Input for updating a user setting
+"""
+input UpdateUserSettingsInput {
```
Comment (jjoyce0510) [inline datahub-graphql-core/src/main/resources/entity.graphql:9095]: or maybe this should be updateUserSettingInput

Btw this diverges from how we do global settings, where we allow you to update a full blob at one time!

File: metadata-models/src/main/pegasus/com/linkedin/identity/CorpUserSettings.pdl [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+ */
+@Aspect = {
+  "name": "corpUserSettings"
+}
+record CorpUserSettings {
+  /**
+   * Flag whether the user should see a homepage with only datasets, charts & dashboards. Intended for users
+   * who have less operational use cases for the datahub tool.
+   */
+  showSimplifiedHomepage: optional boolean
```
Comment (jjoyce0510) [inline metadata-models/src/main/pegasus/com/linkedin/identity/CorpUserSettings.pdl:14]: Can we actually put this under an appearances header? This aspect is _undoubtedly_ going to grow and I don't know if it makes sense to have this at top level. Unless all settings + preferences will remain top level. In GlobalSettings we break out into categories. What do you think?

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/corpuser/mappers/CorpUserMapper.java [hunk 4]
```diff
@@ -52,9 +56,32 @@ public CorpUser apply(@Nonnull final EntityResponse entityResponse) {
...
+    private void mapCorpUserSettings(@Nonnull CorpUser corpUser, EnvelopedAspect envelopedAspect) {
+        CorpUserSettings corpUserSettings = new CorpUserSettings();
+        if (envelopedAspect != null) {
+            corpUserSettings = new CorpUserSettings(envelopedAspect.getValue().data());
+        }
+
+        com.linkedin.datahub.graphql.generated.CorpUserSettings result =
+            new com.linkedin.datahub.graphql.generated.CorpUserSettings();
+
+        AppearanceCorpUserSettings appearanceResult = new AppearanceCorpUserSettings();
```
Comment (jjoyce0510) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/corpuser/mappers/CorpUserMapper.java:74]: Minor personal nit: I woulda called this CorpUserAppearanceSettings

Comment (jjoyce0510) [reply datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/corpuser/mappers/CorpUserMapper.java:74]: (ie at parity with CorpUserEditableProperties.pdl)

File: datahub-graphql-core/src/test/java/com/linkedin/datahub/graphql/resolvers/mutate/UpdateUserSettingResolverTest.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+import org.testng.annotations.Test;
+
+import static com.linkedin.datahub.graphql.TestUtils.*;
+
+
+public class UpdateUserSettingResolverTest {
+
+  private static final String TEST_USER_URN = "urn:li:corpuser:test";
+  @Test
+  public void testGetSuccessExistingDomains() throws Exception {
```
Comment (jjoyce0510) [inline datahub-graphql-core/src/test/java/com/linkedin/datahub/graphql/resolvers/mutate/UpdateUserSettingResolverTest.java:26]: existingDomains? Nit: method name issue

File: datahub-web-react/src/app/home/HomePageRecommendations.tsx [hunk 6]
```diff
@@ -73,11 +74,22 @@ type Props = {
     userUrn: string;
 };
 
+const simpleViewEntityTypes = [
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/home/HomePageRecommendations.tsx:77]: oh boy

File: datahub-web-react/src/app/useGetAuthenticatedUser.tsx [hunk 7]
```diff
@@ -10,7 +10,7 @@ export function useGetAuthenticatedUser(skip?: boolean) {
     if (!userUrn) {
         throw new Error('Could not find logged in user.');
     }
-    const { data, error } = useGetMeQuery({ skip, fetchPolicy: 'cache-first' });
+    const { data, error } = useGetMeQuery({ skip, fetchPolicy: 'cache-and-network' });
```
Comment (jjoyce0510) [inline datahub-web-react/src/app/useGetAuthenticatedUser.tsx:13]: Curious - why is this change necessary now?

File: metadata-service/factories/src/main/resources/application.yml [hunk 8]
```diff
@@ -195,3 +195,6 @@ metadataTests:
 
 siblings:
   enabled: ${ENABLE_SIBLING_HOOK:true} # enable to turn on automatic sibling associations for dbt
+
+featureFlags:
+  simplifiedHomepageDefaultOn: ${SHOW_SIMPLIFIED_HOMEPAGE_BY_DEFAULT:false} # shows a simplified homepage with just datasets, charts and dashboards by default to users. this can be configured in user settings
```
Comment (jjoyce0510) [inline metadata-service/factories/src/main/resources/application.yml:200]: Why name the two fields differently?

Comment (jjoyce0510) [reply metadata-service/factories/src/main/resources/application.yml:200]: simplifiedHomepageDefaultOn vs showSimplifiedHomepageByDefault

**Review by jjoyce0510 — APPROVED** [review]
A few minor questions / nitpicks. Overall looking good

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/featureflags/GraphqlFeatureFlags.java [hunk 9]
```diff
@@ -0,0 +1,19 @@
+package com.linkedin.datahub.graphql.featureflags;
+
+public class GraphqlFeatureFlags {
```
Comment (jjoyce0510) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/featureflags/GraphqlFeatureFlags.java:3]: We already have something called ConfigurationProvider -- why do we need this guy?

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/corpuser/mappers/CorpUserMapper.java [hunk 10]
```diff
@@ -52,9 +56,32 @@ public CorpUser apply(@Nonnull final EntityResponse entityResponse) {
...
+        appearanceResult.setShowSimplifiedHomepage(GraphqlFeatureFlags.getDefaultShowSimplifiedHomepage());
```
Comment (jjoyce0510) [inline datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/corpuser/mappers/CorpUserMapper.java:75]: Did we make it a singleton to avoid having to pass additional context here? It seems like we should be doing that, or setting this field after the mapper has run. 

We can do this by passing an instance of _configurationProvider.featureFlags() into the GraphQL Type class

**Review by jjoyce0510 — CHANGES_REQUESTED** [review]
I think we should consider avoiding introduction of a new GraphQL Feature Flags singleton, and instead go with a convention we've already established with ConfigurationProvider

File: datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/corpuser/mappers/CorpUserMapper.java (hunk 4 above)
Comment (gabe-lyons) [reply datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/types/corpuser/mappers/CorpUserMapper.java:74]: This was to keep parity with global settings naming conventions

File: datahub-web-react/src/app/useGetAuthenticatedUser.tsx (hunk 7 above)
Comment (gabe-lyons) [reply datahub-web-react/src/app/useGetAuthenticatedUser.tsx:13]: cache-first wouldn't call network, so we wouldnt get updated settings

**Review by jjoyce0510 — APPROVED** [review]
LGTM - awesome work!
